### ISSUE TITLE
Filtering highways with access: no attribute

### ISSUE THREAD

**Issue by r0x0r** [openmaptiles/openmaptiles#1066]
OpenMapTiles includes highways that have "access: no" attribute. The scheme omits this attribute and as the result private highways get rendered as normal highways and there is no way to filter them out with style changes.

What would be the best way to omit these highways?

**Comment by TomPohys** [comment]
Hi @r0x0r, thank you for the idea. Access tag is pretty complicated https://wiki.openstreetmap.org/wiki/Key:access, but if we take into consideration just "access: no" then I think it is a nice new feature. PR is very welcome.

**Comment by TomPohys** [comment]
Hi @r0x0r, thank you for the idea. Access tag is pretty complicated https://wiki.openstreetmap.org/wiki/Key:access, but if we take into consideration just "access: no" then I think it is a nice new feature to be added into the schema. PR is very welcome.

**Comment by r0x0r** [comment]
@TomPohys I am new to OMT development, any pointers where to start? I have taken a look at `layers/transportation/transportation.sql` that looks like a suitable place to make this change. Am I on the right track?

**Comment by TomPohys** [comment]
Hi, in `layers/transportation` there is `mapping.yaml` which is recipe for `imposm` what and how to import from `area.pbf` to PostgreSQL database. In this file, there is the creation of [`highway_linestring`](https://github.com/openmaptiles/openmaptiles/blob/295688eb804624a6a3132f81b90391b052af6d3e/layers/transportation/mapping.yaml#L165) table. There is no need to adding new column into table, so it is enough to add new [mapping key](https://github.com/openmaptiles/openmaptiles/blob/295688eb804624a6a3132f81b90391b052af6d3e/layers/transportation/mapping.yaml#L206), which will add a new `key:value` (`'access'=>'no'`) into column `tags` (hstore). So it should be enough to add:

```diff
231      - raceway
232      - construction
233      public_transport:
234      - platform
235      man_made:
236      - pier
+ 237    access:
+ 238     - no
```

after that, you need to run again `make import-osm` to update data in the database. I hope that it is understandable. :smiley:

**Comment by pnorman** [comment]
> Hi @r0x0r, thank you for the idea. Access tag is pretty complicated https://wiki.openstreetmap.org/wiki/Key:access, but if we take into consideration just "access: no" then I think it is a nice new feature to be added into the schema. PR is very welcome.

It's not too hard to handle access logic in SQL. `COALESCE(motor_vehicle, vehicle, access, 'yes')` will handle the access hierarchy for motor vehicles on roads

**Comment by boldtrn** [comment]
Wouldn't it be possible to include all common values the access tag in OMT? That way, ways with `access=destination|private|no...` could be styled accordingly.

**Comment by TomPohys** [comment]
Hi @boldtrn, yes, it is possible to import all common values into PostgreSQL, then create some SQL logic (as @pnorman suggested) and adds one new field for `transportation` layer. The main question is how much will new field increase a final mbtiles size - this has to be tested. 

Another question can be if it is realy necessary to have it in base maps.

**Comment by r0x0r** [comment]
@TomPohys I have tried adding a "access - no" mapping to `layers/transportation` and `import/osm` fails with a `error in mapping file: mapping value '' not a string` error.

The rationale behind this feature is simple. I got an angry email from a property owner, who is upset that every openmaptiles based map displays a road going through their property. The road is private and has no public access, but is displayed on a map as a regular road. I guess I would be pretty upset too, if I lived on that property.

In my opinion the best course would be just to filter out highways with `access: no` attribute, so that they would not be displayed at all. This way it won't have any effect on the  tile size.
